build.gradle.kts

```diff
@@ -104,54 +104,130 @@ extra.apply {
   val baseFilename = "${project.name}-${version}"
   set("targetFilePathBase", "${buildDir}/dist/${baseFilename}")
 
+  val shadowJarFilename = "${baseFilename}-all.jar"
+  set("shadowJarFilename", shadowJarFilename)
+
   val javaHome = System.getProperty("java.home") ?: throw GradleException("java.home is not set")
-  val jpackage = javaHome + File.separator + "bin" + File.separator + "jpackage"
+  val jpackage = "${javaHome}/bin/jpackage"
   val jPackageCmd = if (jpackage.contains(" ")) "\"" + jpackage + "\"" else jpackage
   set("jPackageCmd", jPackageCmd)
-  val sharedParameters = ArrayList<String>(listOf(
+
+  val copyrights = "Copyright ©2001–${SimpleDateFormat("yyyy").format(Date())} ${project.name} developers"
+
+  // Platform-agnostic jpackage parameters shared across all the builds.
+  val params = ArrayList<String>(listOf(
       jPackageCmd,
     "--verbose",
       "--input", "${buildDir}/libs",
       "--main-class", "com.cburch.logisim.Main",
-      "--main-jar", "${baseFilename}-all.jar",
+      "--main-jar", shadowJarFilename,
       "--app-version", version,
-      "--copyright", "\"Copyright ©2001–${SimpleDateFormat("yyyy").format(Date())} ${project.name} developers\"",
+      "--copyright", "\"${copyrights}\"", // Must be quoted for shell happines.
```

Comment:
When you added the quotes to the copyright string, were you addressing a specific failure? Because the result is that the quotes are passed right through to the Info.plist file in the Mac app. There is nothing stripping them out. They are not needed in the Info.plist file since it uses xml style tags: <string>stuff</string>

I also notice that you named the function runShellCommand. However, there is no shell in the process. processBuilder passes the array of parameters directly to a Unix system call (one of the exec calls) without using a shell for anything. I'm not positive what happens on Windows, but I suspect there is no shell equivalent there either.

That is why putting quote marks around the jpackage command causes failure on my Mac (and I strongly suspect, on all Unix systems including Linux).

Reply:
> were you addressing a specific failure?

Yes. 

> They are not needed in the Info.plist file since it uses xml style tags

But quotes are string wrappers only, not a part of it. Should not pass IIRC. I can double check that on macOS/Windows tomorrow.

Windows weren't tested yet.